idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -536,127 +536,125 @@ def initialize(self, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
             solver (str): Solver to use for initialization
             optarg (dict): Solver arguments dictionary
         """
-        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
-        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
 
-        opt = SolverFactory(solver)
-        opt.options = optarg
-
-        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
-            res = opt.solve(self, tee=slc.tee)
-        init_log.info_high(
-                "Initialization Step 1 {}.".format(idaeslog.condition(res))
-            )
+        flags = blk.control_volume.initialize(state_args=state_args,
+                                              hold_state=True)
 
-        for t in self.flowsheet().config.time:
-            for k, v in self.inlet.vars.items():
+        for t in blk.flowsheet().config.time:
+            for k, v in blk.inlet.vars.items():
                 v[t].fix()
-            for k, v in self.outlet.vars.items():
+            for k, v in blk.outlet.vars.items():
                 v[t].unfix()
 
             # if there is not a good guess for efficiency or outlet pressure
             # provide something reasonable.
-            eff = self.efficiency_isentropic[t]
+            eff = blk.efficiency_isentropic[t]
             eff.fix(eff.value if value(eff) > 0.3
                     and value(eff) < 1.0 else 0.85)
             # for outlet pressure try outlet pressure, pressure ratio
             # then if none of those look reasonable use a pressure ratio of 2.0
             # to calculate outlet pressure
-            Pout = self.outlet.pressure[t]
-            Pin = self.inlet.pressure[t]
+            Pout = blk.outlet.pressure[t]
+            Pin = blk.inlet.pressure[t]
             prratio = value(Pout/Pin)
```

Comment:
As it is currently written, this will almost always evaluate to 1, as Pout will be the same as Pin unless the user fixed Pout as their design spec. It could be that you are assuming that this will be the case, but that would need to be clearly documented, and we would need to have checks to make sure the users didn't try fixing deltaP or ratioP instead.

What you really need to do is determine which out of Pout, detalP and ratioP is fixed, and then work from there.